### PR TITLE
[Bug?]: ZipFS extension broken on VSCode 1.66

### LINKED ISSUE #4304: [Bug?]: ZipFS extension broken on VSCode 1.66
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

They break it again™️ (previously: #3539)

```
Unable to open 'index.d.ts': Unable to read file 'zip://users/kid/Developer/react-parcel-play/.yarn/cache/preact-npm-10.7.0-6d31f88d66-fd0c016437.zip/node_modules/preact/src/index.d.ts' (Error: ENOENT: no such file or directory, stat '/kid/Developer/react-parcel-play/.yarn/cache/preact-npm-10.7.0-6d31f88d66-fd0c016437.zip/node_modules/preact/src/index.d.ts').
```

I assume this is more like a VSCode issue rather than TS issue since they bumped the Electron from 13 to 17.

### To reproduce

1. Install ZipFS extension (v2.3.0) on VSCode (1.66.0)
2. Install TypeScript (4.6.3)
3. Enable VSCode SDK via `yarn dlx @yarnpkg/sdks vscode`
4. Install some package with types and try "Go to definition"

### Environment

```shell
System:
    OS: macOS 12.3
    CPU: (8) x64 Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz
  Binaries:
    Node: 17.8.0 - /private/var/folders/x5/9rywd94n2qdbvjs0sj59g4800000gn/T/xfs-bc5e87c9/node
    Yarn: 3.2.0 - /private/var/folders/x5/9rywd94n2qdbvjs0sj59g4800000gn/T/xfs-bc5e87c9/yarn
    npm: 8.5.5 - /usr/local/bin/npm
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ojkelly**
**What's the problem this PR addresses?**
A recent Vscode update broke ZipFS links.

Fixes #4304

...

**How did you fix it?**

It appears that the `vscode.URI.fsPath` helper (described as just [sugar](https://github.com/microsoft/vscode/blob/24e73f3551a59166d34d0deecbb623c24af67645/src/vs/base/common/uri.ts#L202)) no longer prepends the `authority`.

On macOS `/Users/owen/code` became `/owen/code`, which could not be found.

I can't find the specific commit where this change occurred, but this fix does resolve the issue.

I didn't apply this change to the `TerminalLinkProvider`, as I wasn't able to get a good repro setup going for that. I suspect it will need some kind of fix.

...

**Checklist**
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [ ] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/vscode-zipfs/sources/index.ts [hunk 1]
```diff
@@ -1,11 +1,20 @@
-import {npath}                        from '@yarnpkg/fslib';
+import {npath, NativePath}            from '@yarnpkg/fslib';
+import path                           from 'path';
 import * as vscode                    from 'vscode';
 
+
```
Comment (ojkelly) [inline packages/vscode-zipfs/sources/index.ts:5]:
```suggestion
```

**Comment by merceyz** [comment]
Thanks for the PR however the issue is in the SDK, VSCode 1.66 changed the format for in-memory file paths.
I've opened https://github.com/yarnpkg/berry/pull/4330 to update the pattern so it works with VSCode 1.66.

**Comment by JaLe29** [comment]
Great job @merceyz !

Is there any way how to immediately use it?

Or we must wait for next yarn release?

**Comment by arcanis** [comment]
Released as 4.0.0-rc.1 (`yarn dlx @yarnpkg/sdks` should pull it)

**Comment by JaLe29** [comment]
After update sometimes I am facing to the same issue. Any tips? How can I debug (check some log) this issue?

VS code: 1.66.1 (windows 11 64b)
Yarn: 3.2.0

See: 
https://user-images.githubusercontent.com/22296816/162608400-53241a06-c342-4c9f-8ca4-fcb272ece786.mp4

e

**Pull request closed without merging**